grunz/tests/test_file_utils.py

```diff
@@ -0,0 +1,156 @@
+import unittest
+import tempfile
+import os
+import time
+import hashlib
+from pathlib import Path
+from filecmp import cmp
+
+from grunz.file_utils.file_utils import FileUtils
+
+
+class TestFileUtils(unittest.TestCase):
+
+    def setUp(self):
+        # Creates a temporary directory for test files
+        self.test_dir = tempfile.TemporaryDirectory()
+        self.file_utils = FileUtils(Path(self.test_dir.name))
+
+    def tearDown(self):
+        # Cleans up the temporary directory and test files
+        self.test_dir.cleanup()
+
+    def test_find_files_recursively_returning_matching_files(self):
+        """
+        Tests that only the file paths matching the submitted extension are returned.
+        Paths of expected_files are converted to strings with resolved absolute paths to
+        facilitate comparison with result.
+        """
+        self.create_dummy_files('file1.txt', 'file2.txt', 'file3.AVI', 'file4.AVI')
+
+        result = self.file_utils.find_files_recursively('AVI')
+
+        expected_files = [
+            str(Path(self.test_dir.name) / 'file3.AVI'),
+            str(Path(self.test_dir.name) / 'file4.AVI')
+        ]
+        expected_files = [str(Path(p).resolve()) for p in expected_files]
+
+        self.assertEqual(result, expected_files, "Returned files does not match expected files.")
+
+    def test_find_files_recursively_sorting_files(self):
+        """
+        Tests that the returned list of file paths is correctly sorted.
+        Paths of expected_files are converted to strings with resolved absolute
+        paths to facilitate comparison with result.
+        """
+        self.create_dummy_files('file1.txt', 'file4.jpeg', 'file2.jpeg', 'file3.jpeg')
+
+        result = self.file_utils.find_files_recursively('jpeg')
+
+        expected_files = [
```

Comment:
This is a good use case for pytest parametrize. https://docs.pytest.org/en/7.1.x/example/parametrize.html. Pytest is broadly used in the Python space. What do you think about trying to replace unittest with Pytest? Perhaps we could do that in another PR? How do you see it?

Reply:
That does look like it allows you to do more thorough testing without having to write lots more code. I'm not too attached to unittest, I just chose it because it's part of the Python standard library and it seemed easy to use and get the tests I wanted running.

Happy to give pytest a go in another PR, will be good to learn about it!